### ISSUE TITLE
com.hazelcast.internal.serialization.impl.compact.integration.CompactFormatSplitBrainTest.testSplitBrainHealing_whenSmallerClusterHasNoSchemas [API-1996]

### ISSUE THREAD

**Issue by sumnerib** [hazelcast/hazelcast#23191]
_master_ (commit 6108cba3d8622fe56fe98b5edb03796c0dc3359e)

Failed on Windows - MicrosoftJDK17: https://jenkins.hazelcast.com/job/Hazelcast-master-Windows-MicrosoftJDK17/41/testReport/junit/com.hazelcast.internal.serialization.impl.compact.integration/CompactFormatSplitBrainTest/testSplitBrainHealing_whenSmallerClusterHasNoSchemas/
<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: Cluster size is not correct. Expected: 2, actual: 1, instance index: 0
	at org.junit.Assert.fail(Assert.java:89)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSize(HazelcastTestSupport.java:1047)
	at com.hazelcast.test.HazelcastTestSupport.lambda$assertClusterSizeEventually$9(HazelcastTestSupport.java:1082)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1262)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1279)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSizeEventually(HazelcastTestSupport.java:1082)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSizeEventually(HazelcastTestSupport.java:1060)
	at com.hazelcast.internal.serialization.impl.compact.integration.CompactFormatSplitBrainTest.splitCluster(CompactFormatSplitBrainTest.java:171)
	at com.hazelcast.internal.serialization.impl.compact.integration.CompactFormatSplitBrainTest.testSplitBrainHealing_whenSmallerClusterHasNoSchemas(CompactFormatSplitBrainTest.java:82)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:833)

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-Windows-MicrosoftJDK17/41/

**Comment by Patras3** [comment]
Failed again on _5.2.z_ with Windows - OracleJDK8: https://jenkins.hazelcast.com/job/Hazelcast-5.maintenance-Windows-OracleJDK8/97/testReport/junit/com.hazelcast.internal.serialization.impl.compact.integration/CompactFormatSplitBrainTest/testSplitBrainHealing_whenSmallerClusterHasNoSchemas/

**Comment by mdumandag** [comment]
This is the same problem that is fixed in https://github.com/hazelcast/hazelcast/pull/23148

Members of the smaller brain cannot form a cluster because one of the members rejects the other's mastership claim because of the situation explained in the linked pull request.

From the logs, where we expect 5704 and 5705 to form a cluster:

```
10:04:24,391 ERROR |testSplitBrainHealing_whenSmallerClusterHasNoSchemas| - [FetchMembersViewOp] hz.CompactFormatSplitBrainTest_modest_murdock.priority-generic-operation.thread-0 - [127.0.0.1]:5705 [dev] [5.3.0-SNAPSHOT] [127.0.0.1]:5704 claims mastership but it is not a member!
java.lang.IllegalArgumentException: [127.0.0.1]:5704 claims mastership but it is not a member!
	at com.hazelcast.internal.util.Preconditions.checkTrue(Preconditions.java:327) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.ClusterServiceImpl.handleMastershipClaim(ClusterServiceImpl.java:271) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.operations.FetchMembersViewOp.run(FetchMembersViewOp.java:60) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:192) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:295) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:270) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:498) ~[classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:197) ~[classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:137) ~[classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.executeRun(OperationThread.java:123) ~[classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) ~[classes/:?]
```

This fix is also backported to 5.2.z. So, there is nothing extra we need to do. I am closing this issue.
